src/chat/tools/getDockerContainers.ts

```diff
@@ -0,0 +1,150 @@
+import { summarizeLocalDockerContainer } from "../summarizers/dockerContainers";
+import { BaseLanguageModelTool, TextOnlyToolResultPart } from "./base";
+
+const logger = new Logger("chat.tools.getDockerContainers");
+
+export interface IGetDockerContainersParameters {
+  resourceKind: LocalResourceKind;
+}
+
+export class GetDockerContainersTool extends BaseLanguageModelTool<IGetDockerContainersParameters> {
+  readonly name = "get_dockerContainers";
+  readonly progressMessage = "Checking available Docker containers...";
+
+  async invoke(
+    options: LanguageModelToolInvocationOptions<IGetDockerContainersParameters>,
+    token: CancellationToken,
+  ): Promise<LanguageModelToolResult> {
+    const params = options.input;
+    logger.debug("params:", params);
+    // verify we're only trying to look up Kafka/SR containers
+    const validKinds: LocalResourceKind[] = Object.values(LocalResourceKind);
+    if (!validKinds.includes(params.resourceKind)) {
+      return new LanguageModelToolResult([
+        new LanguageModelTextPart(
+          `The resource kind "${params.resourceKind}" is not valid. Select one of ${JSON.stringify(validKinds)} and try again.`,
+        ),
+      ]);
+    }
+
+    // first check if Docker is available, and if not, let the model know what the error is
+    const client = new SystemApi();
+    const init: RequestInit = await defaultRequestInit();
+    try {
+      await client.systemPing(init);
+    } catch (error) {
+      if (error instanceof Error) {
+        return new LanguageModelToolResult([
+          new LanguageModelTextPart(`Unable to connect to the Docker engine: "${error.message}"`),
+        ]);
+      } else {
+        return new LanguageModelToolResult([
+          new LanguageModelTextPart(`Unable to connect to the Docker engine: "${error}"`),
+        ]);
+      }
+    }
+
+    // get the array of ContainerSummary objects first
+    let summaries: ContainerSummary[] = [];
+    switch (params.resourceKind) {
+      case LocalResourceKind.Kafka:
+        summaries = await getLocalKafkaContainers();
+        break;
+      case LocalResourceKind.SchemaRegistry:
+        summaries = await getLocalSchemaRegistryContainers();
+        break;
+    }
+    if (!summaries.length) {
+      return new LanguageModelToolResult([
+        new LanguageModelTextPart(
+          `No Docker containers found for resource kind "${params.resourceKind}".`,
+        ),
+      ]);
+    }
+
+    // if we have at least one ContainerSummary, inspect to get config.env details
+    const inspectPromises: Promise<ContainerInspectResponse>[] = [];
+    summaries.forEach((summary) => {
+      if (summary.Id) {
+        inspectPromises.push(getContainer(summary.Id));
+      }
+    });
+    const containers: ContainerInspectResponse[] = await Promise.all(inspectPromises);
+    if (!containers.length) {
+      // shouldn't happen since we just got summaries
+      return new LanguageModelToolResult([
```

Comment:
I am not seeing a new result in this flow:

1. Set up local SR+Kafka, new chat, ask about my local SR+Kafka, get response. 
2. Tear down local SR+Kafka.
3. Same chat, ask about my local SR+Kafka, get info that it's running. 
I suspect this is something in `participant.ts` not this file. 
<img width="1572" alt="Screenshot 2025-05-14 at 9 23 11 AM" src="https://github.com/user-attachments/assets/1f537405-947f-428c-8836-68238476eb4c" />

Reply:
fixed by not filtering the statuses https://github.com/confluentinc/vscode/pull/1721/files#diff-4d0b5993e2771d6353da797828596c52fe215e0dd2775eb2ff30379ffe9e7748R71